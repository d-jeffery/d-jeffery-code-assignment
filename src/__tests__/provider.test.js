const { Provider } = require('../provider');

describe('Provider', () => {
  test('create with name', () => {
    const provider = new Provider('Dr Jones');

    expect(provider.name).toBe('Dr Jones');
  });

  test('invalid range should throw', () => {
    const provider = new Provider('Dr Jones');

    // startTime > endTime
    const startTime = Date.now() - (60 * 60 * 1000);
    const endTime = Date.now();
    try {
      provider.addAppointments(startTime, endTime);
    } catch (err) {
      expect(err).toBe('Invalid time range');
    }
  });

  test('empty range should have 0 appointments', () => {
    const provider = new Provider('Dr Jones');

    // Range is empty
    const startTime = Date.now();
    provider.addAppointments(startTime, startTime);

    expect(provider.appointments.size).toBe(0);
  });

  test('getAppointments should return consistent times for a given range', () => {
    const provider = new Provider('Dr Jones');

    const startTime = new Date(1723680000000).getTime();
    // Add 4 hours to start time
    const endTime = new Date(1723680000000 + (4 * 60 * 60 * 1000));

    provider.addAppointments(startTime, endTime);

    expect(provider.appointments.size).toBe(16);
    expect(provider.getAppointments()).toMatchSnapshot();
  });

  test('bookAppointments should set confirmed to true', () => {
    const provider = new Provider('Dr Jones');

    const startTime = new Date(1723680000000).getTime();
    // Add 4 hours to start time
    const endTime = new Date(1723680000000 + (60 * 60 * 1000));

    provider.addAppointments(startTime, endTime);

    provider.bookAppointment(startTime);

    expect(provider.appointments.size).toBe(4);
    expect(provider.getAppointments()).toMatchSnapshot();
  });
});
