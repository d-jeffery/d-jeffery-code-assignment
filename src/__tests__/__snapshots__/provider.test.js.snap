// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Provider bookAppointments should set confirmed to true 1`] = `
Map {
  1723680000000 => true,
  1723680900000 => false,
  1723681800000 => false,
  1723682700000 => false,
}
`;

exports[`Provider getAppointments should return consistent times for a given range 1`] = `
Map {
  1723680000000 => false,
  1723680900000 => false,
  1723681800000 => false,
  1723682700000 => false,
  1723683600000 => false,
  1723684500000 => false,
  1723685400000 => false,
  1723686300000 => false,
  1723687200000 => false,
  1723688100000 => false,
  1723689000000 => false,
  1723689900000 => false,
  1723690800000 => false,
  1723691700000 => false,
  1723692600000 => false,
  1723693500000 => false,
}
`;
